polyfill/lib/ecmascript.mjs

```diff
@@ -358,13 +359,20 @@ export function RejectTemporalLikeObject(item) {
   }
 }
 
+export function CanonicalizeTimeZoneOffsetString(offsetString) {
+  const offsetNs = ParseTimeZoneOffsetString(offsetString);
+  return FormatTimeZoneOffsetString(offsetNs);
+}
+
 export function ParseTemporalTimeZone(stringIdent) {
-  const { ianaName, offset, z } = ParseTemporalTimeZoneString(stringIdent);
-  if (ianaName) return GetCanonicalTimeZoneIdentifier(ianaName);
+  const { tzName, offset, z } = ParseTemporalTimeZoneString(stringIdent);
```

Comment:
How far are we taking the alignment? [ParseTemporalTimeString](https://tc39.es/proposal-temporal/#sec-temporal-parsetemporaltimestring) returns a value extracted from the [[TimeZone]] field of a [ParseISODateTime](https://tc39.es/proposal-temporal/#sec-temporal-parseisodatetime) response, which includes fields [[Z]], [[OffsetString]], and [[Name]] (NB: not [[TzName]]).

Reply:
There's another divergence from the spec that I didn't get into in this PR, which is that in the spec, ParseISODateTime doesn't return `[[Name]], [[OffsetString]], [[Z]]` as fields, but instead wraps them into a `[[TimeZone]]` field that's a `{ [[Name]], [[OffsetString]], [[Z]] }` record. The polyfill flattens this record in ParseISODateTime.

If we changed the polyfill's ParseISODateTime to match the spec's structure, then changing ParseISODateTime and ParseTemporalTimeZone to use `name` might be OK. But I was concerned that looking at the polyfill code using ParseISODateTime's flattened record that a plain `name` in that code might be ambiguous, i.e. "Name of what?".  So I'm inclined to leave it as `tzName`, unless we also want to un-flatten ParseISODateTime, which someone else is welcome to do but I didn't see a big advantage of doing here because it won't make the code more readable nor will it make it easier to merge in future changes.